drivers/unix/dir_access_unix.cpp

```diff
@@ -309,7 +309,7 @@ Error DirAccessUnix::change_dir(String p_dir) {
 	// prev_dir is the directory we are changing out of
 	String prev_dir;
 	char real_current_dir_name[2048];
-	getcwd(real_current_dir_name, 2048);
+	ERR_FAIL_COND_V(getcwd(real_current_dir_name, 2048) == 0, ERR_BUG);
```

Comment:
Here, I'm assuming `getcwd` must work. Is there any case where this might fail? If yes, we should do proper handling. Or the following `chdir` will fail.

Reply:
Are you sure that `getcwd` will return `0` on failure? `man getcwd` tells me:
```
RETURN VALUE
       On success, these functions return a pointer to a string containing the pathname of the current working directory.  In the case getcwd() and getwd() this is the  same  value  as
       buf.

       On failure, these functions return NULL, and errno is set to indicate the error.  The contents of the array pointed to by buf are undefined on error.

ERRORS
       EACCES Permission to read or search a component of the filename was denied.

       EFAULT buf points to a bad address.

       EINVAL The size argument is zero and buf is not a null pointer.

       EINVAL getwd(): buf is NULL.

       ENAMETOOLONG
              getwd(): The size of the null-terminated absolute pathname string exceeds PATH_MAX bytes.

       ENOENT The current working directory has been unlinked.

       ENOMEM Out of memory.

       ERANGE The size argument is less than the length of the absolute pathname of the working directory, including the terminating null byte.  You need to allocate a bigger array and
              try again.
```

Reply:
Well,
`On failure, these functions return NULL, and errno is set to indicate the error.  The contents of the array pointed to by buf are undefined on error.`
Here I'm just assuming that `0 == NULL` which is the case on any sensible system.
I could do:
`ERR_FAIL_COND_V(!getcwd(real_current_dir_name, 2048), ERR_BUG);` is you think it's clearer.

Reply:
All in all I think it's a weird API that returns a pointer to a string which is the same as the `buf` argument we're passing... Would ignoring the return value but checking `errno` silence the warning, or aren't compilers clever enough to analyze this?

Reply:
On the other end, `chdir` returns 0 on success, so I have to fix at least one instance of that.

Reply:
Yeah that's why I paused at the `== 0` check initially, as in Unix methods it often means success... but not here. So `== NULL` or `!...` might be better indeed.